app/app/services/sftp_gateway.rb

```diff
@@ -0,0 +1,18 @@
+class SftpGateway
+  attr_reader :url, :user, :password
+  def initialize(options)
+    @url = options[:url]
+    @user = options[:user]
+    @password = options[:password]
+  end
+
+  def upload_data(local_data, remote_file_location)
+    session = Net::SSH.start(url, user, password: password, port: 22)
+    sftp = Net::SFTP::Session.new(session)
+    sftp.connect!
+    sftp.upload! local_data, remote_file_location
```

Comment:
As this method requires `local_data` to be an IO-like object, I might suggest a clearer name:

```suggestion
  def upload_data(local_file, remote_file_location)
    session = Net::SSH.start(url, user, password: password, port: 22)
    sftp = Net::SFTP::Session.new(session)
    sftp.connect!
    sftp.upload! local_file, remote_file_location
```